rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,225 @@
+
+## Train - trip assignment feed
+
+In a future state, Glides will be responsible for light rail trip assignments, and generate a new event feed of train-to-trip assignment events. RTR will use this as its primary source of information 
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## GTFS and TODS schedule data
+
+Light rail schedule data in GTFS and (eventually) TODS files will be produced by `gtfs_creator` based on an export from HASTUS. The HASTUS trips for the lead car will be the basis for GTFS trips, and GTFS trips will take on their IDs. Integrating the schedule for multi-car trips (that is, trips with a trailer) into the TODS data will be defined in a separate discussion about the adoption of TODS. When `gtfs_creator` takes a trip from HASTUS and outputs it with only minor factual corrections, it SHOULD retain the original trip ID. However, if a trip is actively modified as part of the `gtfs_creator` disruption modeling process, its ID MUST be changed so as to not overlap with the HASTUS trip IDs.
+
+## Glides schedule import
+
+Glides will import schedule data from either its own HASTUS export or, in the future, TODS. In either case, trip data is loaded into the `scheduled_trips` table. The HASTUS trip ID will be used directly to populate the `trip_id` column. While working directly with HASTUS, some modifications will be necessary to which IDs we actually use. Currently, our export gets a column called `trp_number`, but the actual, database-unique ID is called `trp_int_number`, and the latter is also the value published by `gtfs_creator`. The significant different here is that cars grouped into one logical trip all have the same `trp_number` generally, but as `trp_int_number` is database-unique, that value will differ. We want the benefits of both combined trip information for multiple cars as well as uniqueness and agreement between systems. To achieve this, we will combine the trips for the different constituent cars and determine the ultimate ID in the following manner:
+
+1. Group the trip entries for each schedule ID by the `trp_number` value.
+1. For each resulting group, take the trip record for the lead car and use its `trp_int_number`.
+1. Use information about the trailer trip for data like the run number, but discard its `trp_int_number`.
+
+## Glides trainsheet edit events
+
+The `com.mbta.ctd.glides.trips_updated` event schema will receive a version bump, modifying the `TripKey` datatype. `TripKey` will be the same for both added and scheduled trips, in either case being represented as an object with `"serviceDate"` and `"tripId"` keys, where the `tripId` is sourced from the `trip_id` column of `scheduled_trips` for a scheduled trip, or the auto-generated Glides ID for an added trip.
+
+## Glides train - trip assignment events
```

Comment:
One point that we've discussed verbally but is unclear as written is that this feed only represents what's happening at the moment, and doesn't represent any planned or predicted assignments in the future.